test/distributed/org/apache/cassandra/distributed/test/auth/MutualTlsCertificateValidityPeriodTest.java

```diff
@@ -0,0 +1,329 @@
+                                     .buildSelfSigned();
+
+        truststorePath = CA.toTempKeyStorePath(tempFolder.getRoot().toPath(),
+                                               SERVER_TRUSTSTORE_PASSWORD.toCharArray(),
+                                               SERVER_TRUSTSTORE_PASSWORD.toCharArray());
+
+
+        CertificateBundle keystore = new CertificateBuilder().subject("CN=Apache Cassandra, OU=ssl_test, O=Unknown, L=Unknown, ST=Unknown, C=Unknown")
+                                                             .addSanDnsName(InetAddress.getLocalHost().getCanonicalHostName())
+                                                             .addSanDnsName(InetAddress.getLocalHost().getHostName())
+                                                             .buildIssuedBy(CA);
+
+        Path serverKeystorePath = keystore.toTempKeyStorePath(tempFolder.getRoot().toPath(),
+                                                              SERVER_KEYSTORE_PASSWORD.toCharArray(),
+                                                              SERVER_KEYSTORE_PASSWORD.toCharArray());
+
+        builder.withConfig(c -> c.set("authenticator.class_name", "org.apache.cassandra.auth.MutualTlsWithPasswordFallbackAuthenticator")
+                                 .set("authenticator.parameters", Collections.singletonMap("validator_class_name", "org.apache.cassandra.auth.SpiffeCertificateValidator"))
+                                 .set("role_manager", "CassandraRoleManager")
+                                 .set("authorizer", "CassandraAuthorizer")
+                                 .set("client_encryption_options.enabled", "true")
+                                 .set("client_encryption_options.require_client_auth", "optional")
+                                 .set("client_encryption_options.keystore", serverKeystorePath.toString())
+                                 .set("client_encryption_options.keystore_password", SERVER_KEYSTORE_PASSWORD)
+                                 .set("client_encryption_options.truststore", truststorePath.toString())
+                                 .set("client_encryption_options.truststore_password", SERVER_TRUSTSTORE_PASSWORD)
+                                 .set("client_encryption_options.require_endpoint_verification", "false")
+                                 .set("client_encryption_options.max_certificate_validity_period", "30d")
+                                 .set("client_encryption_options.certificate_validity_warn_threshold", "5d")
+                                 .with(Feature.NATIVE_PROTOCOL, Feature.GOSSIP));
+        CLUSTER = builder.start();
+
+        configureIdentity();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception
+    {
+        LOGGER.info("[test step : @AfterClass] teardown");
+        if (CLUSTER != null)
+            CLUSTER.close();
+    }
+
+    @Test
+    public void testExpiringCertificate() throws Exception
```

Comment:
Should this be `testValidCertificate`?  From what I can tell this generates a valid certificate and verifies it can connect.

Reply:
I see, expires within a day, should expect it to log something, did you intend to add logging validationh ere?

Reply:
yeah, good idea, I'll add the logging validation.